src/main/java/roomescape/repository/ReservationRepository.java

```diff
@@ -1,21 +1,102 @@
 package roomescape.repository;
 
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Repository;
 import roomescape.domain.Reservation;
+import roomescape.domain.Time;
 
 import java.time.LocalDate;
 import java.time.LocalTime;
 import java.util.List;
 import java.util.Optional;
 
-public interface ReservationRepository {
+@Repository
+public class ReservationRepository {
 
-    Reservation save(Reservation reservation);
+    private static final RowMapper<Reservation> RESERVATION_ROW_MAPPER = (rs, rowNum) ->
+    {
+        Time time = new Time(
+                rs.getLong("time_id"),
+                rs.getTime("time_value").toLocalTime()
+        );
+        return new Reservation(
+                rs.getLong("reservation_id"),
+                rs.getString("name"),
+                rs.getDate("date").toLocalDate(),
+                time
+        );
+    };
 
-    List<Reservation> findAll();
+    private final SimpleJdbcInsert jdbcInsert;
+    private final JdbcTemplate jdbcTemplate;
 
-    Optional<Reservation> findById(long reservationId);
+    public ReservationRepository(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.jdbcInsert = new SimpleJdbcInsert(jdbcTemplate)
+                .withTableName("RESERVATION")
+                .usingGeneratedKeyColumns("id");
+    }
 
-    boolean existsByDateAndTime(LocalDate date, LocalTime time);
+    public Reservation save(final Reservation reservation) {
+        SqlParameterSource parameters = new MapSqlParameterSource()
+                .addValue("name", reservation.getName())
+                .addValue("date", reservation.getDate())
+                .addValue("time_id", reservation.getTime().getId());
 
-    void deleteById(long reservationId);
+        long id = jdbcInsert.executeAndReturnKey(parameters).longValue();
+        return new Reservation(
+                id,
+                reservation.getName(),
+                reservation.getDate(),
+                reservation.getTime()
+        );
+    }
+
+    public List<Reservation> findAll() {
+        String selectAll = """
+                SELECT r.id AS reservation_id, r.name, r.date, t.id AS time_id, t.time AS time_value
+                FROM reservation AS r
+                INNER JOIN time AS t
+                ON r.time_id = t.id
+                """;
+        return jdbcTemplate.query(selectAll, RESERVATION_ROW_MAPPER);
+    }
```

Comment:
(개인적인 의견) 저는 SQL에서 인자가 많아지는 경우에는 개행을 추가하는 걸 선호하긴 합니다. (특히나 join까지 있다면 더욱)

```suggestion
    public List<Reservation> findAll() {
        String selectAll = """
                SELECT 
                    r.id AS reservation_id,
                    r.name,
                    r.date,
                    t.id AS time_id,
                    t.time AS time_value
                FROM reservation AS r
                INNER JOIN time AS t
                ON r.time_id = t.id;
                """;
        return jdbcTemplate.query(selectAll, RESERVATION_ROW_MAPPER);
    }
```